ethers-core/src/utils/hash.rs

```diff
@@ -40,6 +40,13 @@ pub fn keccak256<T: AsRef<[u8]>>(bytes: T) -> [u8; 32] {
     output
 }
 
+/// matches keccak256(abi.encode(....)) pattern in solidity
+///
+/// accepts a borrowed array of type abi::Token
+fn solidity_keccak256(args: &[abi::Token]) -> Bytes {
```

Comment:
the naming might cause confusion, as a developer might now wonder what the difference is between the "keccak" function and this function.

maybe `keccak256_tokens`?

wdyt @prestwich

Reply:
I agree that the name could confuse people. My motivation for naming the function with "solidity" prepended is to match the naming convention used in the Ethers.js package. Is it worthwhile to preserve the naming convention? 
![image](https://user-images.githubusercontent.com/106392730/230617121-73897420-b817-4431-a32a-747fdf0a616e.png)

Reply:
the ethers.js package uses this for `keccak256(abi.encodePacked(...))`, so this will result in a different output from ethers.js. I think `solidityKeccak256` is a poor name in the first place, as it doesn't communicate anything

is there a reason this is a util function and not a provided method on one of the tokenization traits?

Reply:
I'm not convinced it's a worthwhile inclusion. If it were, i'd probably name it `encoded_hashed` or similar to signifiy that there are 2 actions being performed

Reply:
Should I add packing support (with the relevant helpers) + move to trait + rename? I think having both packed and non-packed options could be helpful: non-packed is used when there are multiple dynamic types because it prevents potential hash collisions (with the tradeoff of a little extra gas)